### PR TITLE
Docs markdown linting for title casing

### LINKED ISSUE #5464: docs: Add title casing to markdown linting as part of build system
Per discussion with @Aaronontheweb.

npm package: https://www.npmjs.com/package/markdownlint-rule-titlecase

We can extend the current CI steps with this additional rule, and then do the cleanup. 

I'll do a PR quickly and see if I can tackle it.

### PR DESCRIPTION
**Pull request by SeanKilleen**
Resolves #5464

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
This is Christmas for me. Cleans up one of the biggest eyesores with our documentation.

File: build-system/pr-validation.yaml [hunk 1]
```diff
@@ -48,10 +48,10 @@ jobs:
       - task: Npm@1
         inputs:
           command: "custom"
-          customCommand: "install -g markdownlint-cli"
+          customCommand: "install -g markdownlint-cli markdownlint-rule-titlecase"
```
Comment (Aaronontheweb) [inline build-system/pr-validation.yaml:51]: Very nice

**Comment by SeanKilleen** [comment]
> This is Christmas for me. Cleans up one of the biggest eyesores with our documentation.

Happy to help! I've got some family time now but I'll wrap it up soon.

File: build-system/pr-validation.yaml [hunk 2]
```diff
@@ -48,10 +48,10 @@ jobs:
...
       - task: CmdLine@2
         inputs:
-          script: 'markdownlint "docs/**/*.md"'
+          script: 'markdownlint "docs/**/*.md" --rules "markdownlint-rule-titlecase"'
```
Comment (SeanKilleen) [inline build-system/pr-validation.yaml:54]: ℹ️ FYI, I opened a GitHub issue with a question to see if there's any way to move this info config. No way so far that I can see.

ℹ️ I also double-checked that this doesn't _only_ enable that rule (by locally breaking a different rule.)

**Comment by SeanKilleen** [comment]
@Aaronontheweb OK, I think this will finish shortly and should be good to go at that point. Feel free to ping me any questions/concerns in a review! 👍

**Comment by Aaronontheweb** [comment]
Very, very well done. Already included in your changes in my meta-documentation on https://github.com/akkadotnet/akka.net/pull/5461

**Pull request merged**